bundles/org.openhab.binding.octopusenergy/src/main/java/org/openhab/binding/octopusenergy/internal/handler/OctopusEnergyElectricityMeterPointActions.java

```diff
@@ -0,0 +1,177 @@
+import org.openhab.core.automation.annotation.ActionOutput;
+import org.openhab.core.automation.annotation.RuleAction;
+import org.openhab.core.thing.binding.ThingActions;
+import org.openhab.core.thing.binding.ThingActionsScope;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OctopusEnergyElectricityMeterPointActions} class implements actions for Electricity Meter Points.
+ *
+ * @author Rene Scherer - Initial contribution
+ */
+@ThingActionsScope(name = "octopusenergy")
+@NonNullByDefault
+public class OctopusEnergyElectricityMeterPointActions implements ThingActions {
+
+    private static final PriceOptimiser PO = PriceOptimiser.getInstance();
+
+    private final Logger logger = LoggerFactory.getLogger(OctopusEnergyElectricityMeterPointActions.class);
+
+    private @Nullable OctopusEnergyElectricityMeterPointHandler handler;
+
+    @Override
+    public void setThingHandler(ThingHandler handler) {
+        this.handler = (OctopusEnergyElectricityMeterPointHandler) handler;
+    }
+
+    @Override
+    public @Nullable ThingHandler getThingHandler() {
+        return handler;
+    }
+
+    @RuleAction(label = "optimise", description = "Calculates the optimal time and cost for an activity.")
+    public @Nullable @ActionOutput(name = "lastUpdatedTime", type = "java.time.ZonedDateTime") @ActionOutput(name = "optimisedStartTime", type = "java.time.ZonedDateTime") @ActionOutput(name = "optimisedAverageUnitCost", type = "java.math.BigDecimal") @ActionOutput(name = "requestedDuration", type = "java.time.Duration") @ActionOutput(name = "requestedEndTime", type = "java.time.ZonedDateTime") @ActionOutput(name = "requestedStartTime", type = "java.time.ZonedDateTime") Map<String, Object> optimise(
+            @ActionInput(name = "duration", label = "duration", description = "The duration of the activity.") Duration duration) {
+        try {
+            return toResultMap(PO.optimiseWithAbsoluteStartTime(duration, ZonedDateTime.now(), getPriceList()));
+        } catch (NotEnoughDataException e) {
+            logger.warn("Not enough data available for optimizer - {}", e.getMessage());
+        } catch (RecordNotFoundException e) {
+            logger.warn("No Meter Point found for optimizer - {}", e.getMessage());
+        }
+        return new HashMap<String, Object>();
+    }
+
+    public static @Nullable Map<String, Object> optimise(@Nullable ThingActions actions, Duration duration) {
+        if (actions instanceof OctopusEnergyElectricityMeterPointActions) {
+            return ((OctopusEnergyElectricityMeterPointActions) actions).optimise(duration);
+        } else {
+            throw new IllegalArgumentException("Instance is not an OctopusEnergyElectricityMeterPointActions class.");
+        }
+    }
+
+    @RuleAction(label = "optimiseWithRecurringEndTime", description = "Calculates the optimal time and cost for an activity.")
```

Comment:
The label should be human readable and integrateable into a whole sentence like "get optimum end time for a recurring event"